### PR TITLE
Weld integration

### PR DESCRIPTION
**Pull request by ankathur**
Submitting first cut of Weld 3 integration for review. 
Some of the copyrights are yet to be corrected. This change passes the QL and  CDI devtests while TCK has 19 failures.

### CODE REVIEW

**Comment by ankathur** [comment]
@glassfishrobot Run CI tests please

**Comment by ankathur** [comment]
@glassfishrobot Run CI tests please

**Review by jjsnyder — COMMENTED** [review]
I don't understand this.  Why is the new packager/external/osgi-cdi-repackaged needed?  I don't see a similar repackager in previous versions of GlassFish.

**Review by jjsnyder — COMMENTED** [review]
Yes the weld-se-core.jar should be replaced with weld-se-shaded.jar.

**Review by jjsnyder — COMMENTED** [review]
The rest of the changes look good to me.

File: main/appserver/extras/embedded/build.xml [hunk 1]
```diff
@@ -102,7 +102,7 @@
 	  <!-- skip the domain creation template jar files viz., nucleus-domain.jar, appserver-domain.jar -->
 	  <fileset dir="${gfdir}" includes="**/appserver-domain.jar"/>
 	  <fileset dir="${gfdir}" includes="**/nucleus-domain.jar"/>
-	  <fileset dir="${gfdir}" includes="**/weld-se-core.jar"/>
+	  <fileset dir="${gfdir}" includes="**/weld-se-shaded.jar"/>
 	</delete>
 	<delete file="${modulesdir}/autostart/osgi-cdi.jar" failonerror="false"/>
```
Comment (yaminikb) [inline main/appserver/extras/embedded/build.xml]: Jar name should be changed.

File: main/appserver/packager/glassfish-jcdi/pom.xml [hunk 2]
```diff
@@ -95,6 +95,10 @@
             <artifactId>weld-osgi-bundle</artifactId>
         </dependency> 
         <dependency>
+            <groupId>org.jboss.classfilewriter</groupId>
```
Comment (yaminikb) [inline main/appserver/packager/glassfish-jcdi/pom.xml]: I didn't understand the need for classfilewriter changes. You removed the package exports from the weld-integration-fragment pom but bringing in dependency explicitly, why?

File: main/appserver/packager/glassfish-osgi/pom.xml [hunk 3]
```diff
@@ -119,9 +119,9 @@
            <optional>true</optional>
         </dependency>
         <dependency>
-            <groupId>org.glassfish.fighterfish</groupId>
-            <artifactId>osgi-cdi</artifactId>
-            <optional>true</optional>
```
Comment (yaminikb) [inline main/appserver/packager/glassfish-osgi/pom.xml]: You made it non-optional, is that intentional?

Comment (ankathur) [inline main/appserver/packager/glassfish-osgi/pom.xml]: it was unintentional, i will correct that.

File: main/appserver/packager/glassfish-jcdi/pom.xml (hunk 2 above)
Comment (ankathur) [inline main/appserver/packager/glassfish-jcdi/pom.xml]: The jboss-classfilewriter dependency, which is used by Weld for runtime bytecode generation, is no longer bundled within the Weld OSGi bundle. Instead, this dependency needs to be deployed separately to the OSGi container.
This change was introduced in Weld 2.3.

**Comment by ankathur** [comment]
@glassfishrobot Run CI tests please

**Review by yaminikb — APPROVED** [review]

**Review by romain-grecourt — APPROVED** [review]
LGTM

**Pull request merged**